src/orders/views/OrderTransactionRefundEdit/OrderTransactionRefundEdit.tsx

```diff
@@ -80,26 +81,19 @@ const OrderTransactionRefund: React.FC<OrderTransactionRefundProps> = ({ orderId
 
     const { amount, reason, linesToRefund, includeShipping, transactionId } = submitData;
 
-    const dirtyLinesToRefund = linesToRefund.filter(item => item.isDirty);
-
-    const toAdd = dirtyLinesToRefund.map(line => ({
-      quantity: line.quantity,
-      reason: line.reason,
-      id: data.order!.lines[line.row].id,
-    }));
-
-    const toRemove =
-      draftRefund.lines?.reduce<string[]>((acc, line) => {
-        dirtyLinesToRefund.forEach(qty => {
-          const orderLine = data.order!.lines[qty.row];
-
-          if (line.orderLine.id === orderLine.id && !acc.includes(line.id)) {
-            acc.push(line.id);
-          }
+    const draftRefundLines = draftRefund.lines ?? [];
+    const toRemove = draftRefundLines.map(line => line.id);
+    const toAdd = linesToRefund.reduce<OrderGrantRefundUpdateLineAddInput[]>((acc, line, ix) => {
+      if (typeof line.quantity === "number" && line.quantity > 0) {
+        acc.push({
+          id: data.order!.lines[ix].id,
+          quantity: line.quantity,
+          reason: line.reason,
         });
+      }
 
-        return acc;
-      }, []) ?? [];
+      return acc;
+    }, []);
```

Comment:
Suggestion: It seems like we can just `filter` and/or `map` on `linesToRefund` here